src/main.rs

```diff
@@ -256,8 +256,8 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
                 ingredient_count_msg,
                 items_map
                     .get(ingredient_id)
-                    .map(|item| item.name.as_ref())
-                    .unwrap_or("???"),
+                    .map(|item| item.to_string())
+                    .unwrap_or("???".to_string()),
```

Comment:
Found a single function to replace both calls; `map_or_else`; `map_or_else<U, D, F>(self, default: D, f: F) -> U`. It's a mapping function that returns a `U` instead of an `Option<U>` and allows selection of a default via closure.

I don't find it _terribly_ readable, but I can guess why they put the default first; defaults usually take less code and are less likely to span multiple lines, and parameters after multi-line closures can be easy to overlook.